### PR TITLE
Relationship filter

### PR DESCRIPTION
**Pull request by Tweoss**
Uses new relationship filter to check kind of relation, src id, and target

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -8,10 +8,12 @@ require (
...
-	github.com/onosproject/onos-api/go v0.7.45
+	github.com/onosproject/onos-api/go v0.7.52
 	github.com/onosproject/onos-lib-go v0.7.12
 	github.com/onosproject/onos-test v0.6.5
 	github.com/smartystreets/assertions v1.0.0 // indirect
 	github.com/stretchr/testify v1.7.0
 	google.golang.org/grpc v1.33.2
 )
+
+replace github.com/onosproject/onos-api/go => /Users/francischua/src/github.com/onosproject/onos-api/go
```
Comment (adibrastegarnia) [inline go.mod:19]: Remove replace and update topo to use v0.7.64

**Comment by adibrastegarnia** [comment]
I think you need to run go mod tidy and push changes in PR to pass Jenkins job:

https://jenkins.aetherproject.org/job/verify_onos-topo_make-test/88/console

**Comment by Tweoss** [comment]
I can't view Jenkins, but I'll run go mod tidy

File: pkg/store/filter.go [hunk 2]
```diff
@@ -16,6 +16,10 @@ package store
 
 import topoapi "github.com/onosproject/onos-api/go/onos/topo"
 
+func matchRelation(object *topoapi.Object, filter *topoapi.RelationFilter, entities map[topoapi.ID]*topoapi.Object) bool {
+	return object.Type == topoapi.Object_RELATION && string(object.GetRelation().KindID) == filter.GetRelationKind() && string(object.GetRelation().GetSrcEntityID()) == filter.SrcId && (filter.TargetKind != "" || entities[object.GetRelation().GetTgtEntityID()].GetKind().GetName() == filter.TargetKind)
+}
```
Comment (adibrastegarnia) [inline pkg/store/filter.go:21]: The last condition is filter.TargetKind != "" || entities[object.GetRelation().GetTgtEntityID()].GetKind().GetName() == filter.TargetKind). What you are trying to achieve? The idea is to ignore Target ID thing?

Comment (tomikazi) [reply pkg/store/filter.go:21]: The idea is if the target kind ID is specified, then we only want those kinds of objects. It's an optional secondary filter.

Comment (Tweoss) [reply pkg/store/filter.go:21]: Yes, if the filter target kind is not specified, then it shouldn't matter, but if it is specified the objects (relation)'s target kind should be the same as the filter kind.

File: pkg/store/store.go [hunk 3]
```diff
@@ -175,12 +176,57 @@ func (s *atomixStore) List(ctx context.Context, filters *topoapi.Filters) ([]top
...
+					if map_entity, found := entitiesToGet[ep.ID]; found {
+						if filter.TargetKind == "" || ep.GetKind().Name == filter.TargetKind {
+							*map_entity = *ep
+						}
+					}
+				}
+			}
+		}
+		// iterate over entities to make sure we did not miss any valid ones (due to the corresponding relationship being seen first)
+		for id := range entitiesToGet {
```
Comment (tomikazi) [inline pkg/store/store.go:212]: You can do `for id, entity := range ....` then you can just look at entity == nil instead of map test.

**Review by tomikazi — CHANGES_REQUESTED** [review]

**Review by tomikazi — APPROVED** [review]

**Pull request merged**